src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,65 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Object clazzInstance = createNewInstance(clazz);
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzInstance, getInstance(field.getType()));
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Failed to initialize field " + field.getName());
+                }
+            }
+        }
+        return clazzInstance;
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> interfaceImplementationMap = new HashMap<>();
+        interfaceImplementationMap.put(FileReaderService.class, FileReaderServiceImpl.class);
+        interfaceImplementationMap.put(ProductParser.class, ProductParserImpl.class);
+        interfaceImplementationMap.put(ProductService.class, ProductServiceImpl.class);
+        return interfaceImplementationMap.get(interfaceClazz);
```

Comment:
if u receive like a parameter not interface u will get null , because your keys in a Map is interfaces , let's make a check isInterface()